node-manager/src/node.rs

```diff
@@ -107,6 +126,25 @@ pub(crate) async fn connect_peer_if_necessary(
     Ok(())
 }
 
+pub(crate) fn create_peer_manager(
+    km: Arc<KeysManager>,
+    lightning_msg_handler: MessageHandler,
+    logger: Arc<MutinyLogger>,
+) -> PeerManager {
+    let current_time = instant::now();
+    let mut ephemeral_bytes = [0u8; 32];
+    getrandom::getrandom(&mut ephemeral_bytes).expect("Failed to generate entropy");
+
+    PeerManager::new(
+        lightning_msg_handler,
+        km.get_node_secret(Recipient::Node).unwrap(),
```

Comment:
I'd prefer an expect here if this is an unrecoverable sort of error (getrandom is a great example). otherwise error handling plz